com.unity.render-pipelines.universal/Samples~/RenderPassSamples/BlurryRefraction/KawaseBlur.cs

```diff
@@ -0,0 +1,114 @@
+}
+class KawaseBlur : ScriptableRenderPass
+{
+    string profilerTag;
+
+    int tmpId1;
+    int tmpId2;
+
+    RenderTargetIdentifier tmpRT1;
+    RenderTargetIdentifier tmpRT2;
+
+    RenderTargetIdentifier cameraColorTexture;
+    KawaseBlurSettings settings;
+
+    public KawaseBlur(string profilerTag, KawaseBlurSettings settings)
+    {
+        this.profilerTag = profilerTag;
+        this.settings = settings;
+        renderPassEvent = settings.renderPassEvent;
+    }
+
+    // This method is called before executing the render pass.
+    // It can be used to configure render targets and their clear state. Also to create temporary render target textures.
+    // When empty this render pass will render to the active camera render target.
+    // You should never call CommandBuffer.SetRenderTarget. Instead call <c>ConfigureTarget</c> and <c>ConfigureClear</c>.
+    // The render pipeline will ensure target setup and clearing happens in a performant manner.
+    public override void OnCameraSetup(CommandBuffer cmd, ref RenderingData renderingData)
+    {
+        var width = renderingData.cameraData.cameraTargetDescriptor.width / settings.downsample;
+        var height = renderingData.cameraData.cameraTargetDescriptor.height / settings.downsample;
+
+        tmpId1 = Shader.PropertyToID("tmpBlurRT1");
+        tmpId2 = Shader.PropertyToID("tmpBlurRT2");
+        cmd.GetTemporaryRT(tmpId1, width, height, 0, FilterMode.Bilinear, RenderTextureFormat.ARGB32);
+        cmd.GetTemporaryRT(tmpId2, width, height, 0, FilterMode.Bilinear, RenderTextureFormat.ARGB32);
+
+        tmpRT1 = new RenderTargetIdentifier(tmpId1);
+        tmpRT2 = new RenderTargetIdentifier(tmpId2);
+
+        ConfigureTarget(tmpRT1);
+        ConfigureTarget(tmpRT2);
+    }
+
+    // Here you can implement the rendering logic.
+    // Use <c>ScriptableRenderContext</c> to issue drawing commands or execute command buffers
+    // https://docs.unity3d.com/ScriptReference/Rendering.ScriptableRenderContext.html
+    // You don't have to call ScriptableRenderContext.submit, the render pipeline will call it at specific points in the pipeline.
+    public override void Execute(ScriptableRenderContext context, ref RenderingData renderingData)
+    {
+        cameraColorTexture = ((UniversalRenderPipelineAsset)GraphicsSettings.currentRenderPipeline).scriptableRenderer.cameraColorTarget;
+        CommandBuffer cmd = CommandBufferPool.Get(profilerTag);
+
+        RenderTextureDescriptor opaqueDesc = renderingData.cameraData.cameraTargetDescriptor;
+        opaqueDesc.depthBufferBits = 0;
+
+        // first pass
+        // cmd.GetTemporaryRT(tmpId1, opaqueDesc, FilterMode.Bilinear);
+        cmd.SetGlobalFloat("_offset", 1.5f);
+        cmd.Blit(cameraColorTexture, tmpRT1, settings.blurMaterial);
+
+        for (var i = 1; i < settings.blurPasses - 1; i++)
+        {
+            cmd.SetGlobalFloat("_offset", 0.5f + i);
+            cmd.Blit(tmpRT1, tmpRT2, settings.blurMaterial);
```

Comment:
This won't work on XR. @thomas-zeng can you help?

Reply:
XR prefers to use DrawMesh will fullscreen quad to do blit.
cmd.Blit in some cases will use built-in blit shader which doesn't work well with URP. 
With DrawMesh, we can explicit set the blit shader. See how URP SSAO handles blit here: 
https://github.com/Unity-Technologies/Graphics/blob/1d37e201e0353347878642680c61e7012b156e8f/com.unity.render-pipelines.universal/Runtime/RendererFeatures/ScreenSpaceAmbientOcclusion.cs#L289

Reply:
The result is used in Shader Graph is it possible to use XR textures in shader graph?